File: programmers/Level1/CPP/문자열을 정수로 바꾸기.cpp
#include <string>
using namespace std;

int solution(string s) {
    int answer = 0;
    
    for(int i = 0; i<s.size();i++){
        if(s[i] == '-' || s[i] == '+'){
            continue;
        }
        answer *= 10;
        answer += s[i]-'0';
    }
    
    if(s[0] == '-'){
            answer = 0 - answer;
    }
    
    return answer;
}

File: programmers/Level1/JAVA/문자열을 정수로 바꾸기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        
        if(s.charAt(0) == '-'){
            for(int i =1; i<s.length(); i++){
                answer *= 10;
                answer += s.charAt(i) - '0';
            }
            answer = 0-answer;
        }
        else if(s.charAt(0) == '+'){
            for(int i = 1; i<s.length(); i++){
                answer *= 10;
                answer += s.charAt(i) - '0';
            }
        }
        else {
            for(int i =0; i<s.length(); i++){
                answer *= 10;
                answer += s.charAt(i) - '0';
            }
        }
        
        return answer;
    }
}
